src/hooks/useAnnouncer/useAnnouncer.ts

```diff
@@ -0,0 +1,69 @@
+import { useRef } from "react";
+
+/* Due to possibility of multiple Carbon versions being used on the page,
+ * it is neccessary to maintain the same values for the following ids */
+const POLITE_REGION_ID = "__CARBON_INTERVALS_POLITE_LIVE_REGION";
+const ASSERTIVE_REGION_ID = "__CARBON_INTERVALS_ASSERTIVE_LIVE_REGION";
+
+const TIMEOUT_DURATION = 200;
+type Assertiveness = "polite" | "assertive";
+
+function getRegion(assertiveness: Assertiveness): HTMLElement {
+  const id =
+    assertiveness === "assertive" ? ASSERTIVE_REGION_ID : POLITE_REGION_ID;
+  const region = document.getElementById(id);
+
+  if (region) return region;
+
+  const element: HTMLDivElement = document.createElement("div");
+  const visuallyHiddenStyles: Partial<CSSStyleDeclaration> = {
+    clip: "rect(0 0 0 0)",
+    clipPath: "inset(50%)",
+    height: "1px",
+    overflow: "hidden",
+    position: "absolute",
+    whiteSpace: "nowrap",
+    width: "1px",
+  };
+  element.setAttribute("id", id);
+  element.setAttribute(
+    "role",
+    assertiveness === "assertive" ? "alert" : "status"
+  );
+  element.setAttribute("aria-live", assertiveness);
+  element.setAttribute("aria-atomic", "false");
+  element.setAttribute("aria-relevant", "additions");
+  Object.assign(element.style, visuallyHiddenStyles);
+
+  return document.body.appendChild(element);
+}
+
+function useAnnouncer(): {
```

Comment:
Currently, I have faced some issues with repeated screen reader announcements with VoiceOver and NVDA. I have a feeling it may be due the hook not cleaning itself up properly - needing to use an effect with a returned cleanup to remove the DOM elements on unmount, something like this:

```tsx
useEffect(() => () => destroyLiveRegions());
```

However, I haven't been able to verify this and encountering more problems with setting this up, especially since we need the live regions available in the DOM for the various micro-frontends to use.

Reply:
I agree with @robinzigmond about seeing if we can avoid it using a custom hook. However, if we do and end up needing to use a useEffect we should either use the `useStableCallback` or a similar approach that stores the announcer functions in refs if for some reason it doesn't work

Reply:
Thanks for this guys 👍🏼 I've scratching my head for a while on this one. I like the idea of using two separate hooks for `announcePolitely` and `announceAssertively` instead